testing/fpgaconf/test_fpgaconf_c.cpp

```diff
@@ -679,6 +688,54 @@ TEST_P(fpgaconf_c_p, main2) {
   EXPECT_NE(fpgaconf_main(6, argv), 0);
 }
 
+/**
+ * @test       embed_nullchar
+ * @brief      Test: fpgaconf_main
+ * @details    When the command params contains nullbyte,<br>
+ *             fpgaconf_main displays an error and returns non-zero.<br>
+ */
+TEST_P(fpgaconf_c_p, embed_nullchar1) {
+  copy_bitstream("copy_bitstream.gbs");
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  char four[20];
+  char five[20];
+  strcpy(zero, "fpgaconf");
+  strcpy(one, "-v");
+  strcpy(two, "-n");
+  strcpy(three, "-B");
+  strcpy(four, "0x5e");
+  strcpy(five, "copy_bitstream\0.gbs");
+
+  char *argv[] = { zero, one, two, three, four,
+                   five };
+  EXPECT_NE(fpgaconf_main(6, argv), 0);
```

Comment:
Right 👍 , I would expect not-equal to 0, hopefully because the app realizes that "copy_bitstream" doesn't exist.